### PR TITLE
[FS] Order GetWithdrawals by UTXO usage order

### PR DESCRIPTION
**Pull request by quantumagi**
Ensure that this list is ordered as expected and would be the same on peer nodes. Currently the ordering can differ on peer nodes.

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs [hunk 1]
```diff
@@ -801,22 +801,15 @@ public bool RemoveTransientTransactions(uint256 depositId = null)
...
-            }
+        private OutPoint EarliestOutput(Transaction transaction)
+        {
+            var comparer = Comparer<OutPoint>.Create((x, y) => this.CompareOutpoints(x, y));
+            return transaction.Inputs.Select(i => i.PrevOut).OrderBy(t => t, comparer).FirstOrDefault();
         }
 
         /// <inheritdoc />
-        public List<(Transaction, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null)
+        public List<(Transaction, IWithdrawal)> FindWithdrawalTransactions(uint256 depositId = null, bool sort = false)
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: Will we ever want to find `FindWithDrawalTransactions` unordered?

File: src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs [hunk 2]
```diff
@@ -877,16 +877,18 @@ private bool TransactionHasValidUTXOs(Transaction transaction, List<Coin> coins
...
 
-        /// <inheritdoc />
-        public int CompareOutpoints(OutPoint outPoint1, OutPoint outPoint2)
+        /// <summary>
+        /// Compares two outpoints to see which occurs earlier.
+        /// </summary>
+        /// <param name="outPoint1">The first outpoint to compare.</param>
+        /// <param name="outPoint2">The second outpoint to compare.</param>
+        /// <returns><c>-1</c> if the <paramref name="outPoint1"/> occurs first and <c>1</c> otherwise.</returns>
+        internal int CompareOutpoints(OutPoint outPoint1, OutPoint outPoint2)
```
Comment (fassadlr) [inline src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: I see that `CompareOutpoints` is only ever called from CCTS so why not just move it there?

Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: Its used by `FindWithdrawalTransactions`.

File: src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs (hunk 1 above)
Comment (quantumagi) [reply src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs]: Yes, it is used in that way.

**Pull request merged**